source/dol/bases/d_ac_py_key.cpp

```diff
@@ -0,0 +1,355 @@
+    mTriggeredButtons = dGameKey_c::m_instance->keys[mRemoconID]->mTriggeredButtons;
+}
+
+bool dAcPyKey_c::checkHipAttack() {
+    mHipAtkRead = true;
+    if (!mHipAtkHoldStart) {
+        if (triggerDown()) {
+            mHipAtkHoldStart = true;
+            mHipAtkHoldCount = 0;
+        }
+    } else if (buttonDown()) {
+        if (buttonHipAttack() && ++mHipAtkHoldCount >= 4) {
+            return true;
+        }
+    } else {
+        mHipAtkHoldStart = false;
+    }
+    return false;
+}
+
+void dAcPyKey_c::onStatus(u16 status) {
+    switch (status) {
+        case STATUS_FORCE_JUMP:
+            offStatus(STATUS_FORCE_NO_JUMP);
+            break;
+        case STATUS_FORCE_NO_JUMP:
+            offStatus(STATUS_FORCE_JUMP);
+            break;
+        case STATUS_NO_INPUT:
+        case STATUS_DEMO:
+        case STATUS_FORCE_RIGHT:
+            clearAllKey();
+            break;
+        case STATUS_DISABLE_LR:
+            clearCrossKey();
+            break;
+        default: break;
+    }
+    mStatus |= (1 << status);
+}
+
+void dAcPyKey_c::offStatus(u16 status) {
+    mStatus &= ~(1 << status);
+}
+
+void dAcPyKey_c::clearAllKey() {
+    mDownButtons = 0;
+    mTriggeredButtons = 0;
+    mDownButtonsDemo = 0;
+    mTriggeredButtonsDemo = 0;
+    mActiveShakeJump = 0;
+    mTriggerShakeJump = 0;
+}
+
+void dAcPyKey_c::clearCrossKey() {
+    mDownButtons &= ~(BUTTON_LEFT | BUTTON_RIGHT);
+    mTriggeredButtons &= ~(BUTTON_LEFT | BUTTON_RIGHT);
+}
+
+int dAcPyKey_c::triggerA() const {
+    return mTriggeredButtons & BUTTON_A;
+}
+
+int dAcPyKey_c::buttonCross() const {
+    return mDownButtons & BUTTON_CROSS;
+}
+
+int dAcPyKey_c::triggerCross() const {
+    return mTriggeredButtons & BUTTON_CROSS;
+}
+
+int dAcPyKey_c::buttonUp() const {
+    return mDownButtons & BUTTON_UP;
+}
+
+int dAcPyKey_c::buttonDown() const {
+    return mDownButtons & BUTTON_DOWN;
+}
+
+int dAcPyKey_c::buttonLeft() const {
+    return mDownButtons & BUTTON_LEFT;
+}
+
+int dAcPyKey_c::buttonRight() const {
+    return mDownButtons & BUTTON_RIGHT;
+}
+
+int dAcPyKey_c::triggerUp() const {
+    return mTriggeredButtons & BUTTON_UP;
+}
+
+int dAcPyKey_c::triggerDown() const {
+    return mTriggeredButtons & BUTTON_DOWN;
+}
+
+int dAcPyKey_c::triggerLeft() const {
+    return mTriggeredButtons & BUTTON_LEFT;
+}
+
+int dAcPyKey_c::triggerRight() const {
+    return mTriggeredButtons & BUTTON_RIGHT;
+}
+
+int dAcPyKey_c::triggerTwo() const {
+    return mTriggeredButtons & BUTTON_TWO;
+}
+
+int dAcPyKey_c::buttonTwo() const {
+    return mDownButtons & BUTTON_TWO;
+}
+
+int dAcPyKey_c::triggerOne() const {
+    if (dGameKey_c::m_instance->keys[mRemoconID]->mAttachedExtension == 1) {
+        return mTriggeredButtons & BUTTON_B;
+    } else {
+        return mTriggeredButtons & BUTTON_ONE;
+    }
+}
+
+int dAcPyKey_c::buttonOne() const {
+    if (dGameKey_c::m_instance->keys[mRemoconID]->mAttachedExtension == 1) {
```

Comment:
Maybe use an enum for the attached extension? Pretty sure this is the Nunchuk but there are other values for sure, like the Classic Controller